com.microsoft.mrtk.spatialmanipulation/BoundsControl/BoundsHandleInteractable.cs

```diff
@@ -101,9 +133,27 @@ protected virtual void LateUpdate()
                 colliders[0].enabled = !IsOccluded;
             }
 
-            // Maintain global scale of the handle(s).
+            // Maintain the aspect ratio/proportion of the handles, globally.
             transform.localScale = Vector3.one;
-            transform.localScale = new Vector3(1.0f / transform.lossyScale.x, 1.0f / transform.lossyScale.y, 1.0f / transform.lossyScale.z);
+            transform.localScale = new Vector3(1.0f / transform.lossyScale.x,
+                                               1.0f / transform.lossyScale.y,
+                                               1.0f / transform.lossyScale.z);
+            
+            // If we don't want to maintain the overall *size*, we scale
+            // by the maximum component of the box so that the handles grow/shrink
+            // with the overall box manipulation.
+            if (!maintainGlobalSize)
+            {
+                transform.localScale = transform.localScale * (MaxComponent(transform.parent.lossyScale) / initialParentScale);
```

Comment:
I think this could work, but imagine the following scenario:
We want to manipulate a cube with scale 2, 1, 1. When we manipulate it in a non-uniform way to change its scale to 2, 1.9, 1, the scale of the handles will remain the same even when we set `maintainGlobalSize` to false because `MaxComponent` remains to be 2. Is this intended? I could see us calculating the scale another way to take the changes to all three dimensions into account.

Reply:
So, there's two quantities at play here, the overall _size_ of the handle (which is controlled by the `maintainGlobalSize` option), and the "aspect ratio" of the handle (i.e., the relative proportion of each "axis" of the global lossy scale), which is always constant.

Regardless of the combination of inputs and options, we must always keep the _aspect ratio_ of the handles constant; i.e., they must never distort. On the other hand, the checkbox lets the user determine whether they want the overall size of the handles to be constant or not.

I use MaxComponent instead of taking like, the average of the scale axes, because of the 2D slate case. The bounds can always, at any time, be flattened to a 2D slate, in which case the shrunken axis can't be taken into account for the handle size. The handle size should always be at least relative to the two largest axes. I somewhat arbitrarily chose to only take the largest axis into consideration because it

- Made the code simpler
- Made it look less weird when scaling with handles along a single axis (i.e., the handle size would only change once both non-flattened axes got smaller, instead of the handles continuously changing size as the 2D area was scaled along a single axis.)

Reply:
OK I think that makes sense, did not really think about the 2D slate case.